test/scripts/region_loading-01.js

```diff
@@ -0,0 +1,45 @@
+// Test basic logpoint functionality. When logpoints are added,
+Test.describe(
+  `Unloading a region should suppress logpoint results.`,
+  async () => {
+    const { assert } = Test;
+
+    const loadTimerange = [0, 5000];
+    const unloadTimerange = [0, loadTimerange[1] / 2];
+    const totalMessageCount = 50;
+
+    await Test.selectSource("doc_rr_region_loading.html");
+    await Test.addBreakpoint("doc_rr_region_loading.html", 20, undefined, {
+      logValue: `"Part1Logpoint Number " + number`,
+    });
+
+    await Test.selectConsole();
+
+    let messages = await Test.waitForMessageCount("Part1Logpoint", totalMessageCount);
+    assert(!Test.findMessages("Loading").length);
+    assert(messages.length == totalMessageCount);
+    await Test.removeAllBreakpoints();
+
+    // Unload a region of the recording 1 second in, 2 seconds in duration.
+    await Test.unloadRegion(...unloadTimerange);
+
+    await Test.addBreakpoint("doc_rr_region_loading.html", 20, undefined, {
+      logValue: `"Part2Logpoint Number " + number`,
+    });
+
+    Test.waitUntil(() => Test.findMessages("Loading...").length === 0);
```

Comment:
I think you forgot the `await` here. Furthermore, if the "Loading..." messages aren't added synchronously, this may finish immediately (before any breakpoint messages are added to the console instead of after all messages are updated).